pkgs/native_assets_cli/example/build/linked_library/hook/build.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:logging/logging.dart';
+import 'package:native_assets_cli/native_assets_cli.dart';
+import 'package:native_toolchain_c/native_toolchain_c.dart';
+
+void main(List<String> args) async {
+  await build(args, (config, output) async {
+    final logger = Logger('')
+      ..level = Level.ALL
+      ..onRecord.listen((record) => print(record.message));
+
+    final mathBuilder = CBuilder.library(
```

Comment:
Maybe structure as `final builders = [`

And then a for loop that runs them in sequence with a comment that they need to run in sequence.

(This is more of the design to have all the config in the beginning of the file, and all execution afterwards.)